tests/e2e/vllm/test_train.py

```diff
@@ -0,0 +1,88 @@
+import subprocess
+from pathlib import Path
+
+import numpy as np
+import pytest
+import torch
+from huggingface_hub import snapshot_download
+from loguru import logger
+
+from speculators.data_generation.vocab_mapping import (
+    build_vocab_mappings_from_distribution,
+)
+from tests.e2e.vllm.utils import run_vllm_engine
+
+
+class TestTrainvLLM:
+    """
+    An e2e test which trains a speculator model using pre-computed hidden states
+    and runs the trained model in vLLM.
+    """
+
+    def _generate_t2d_d2t(self, token_freq: str, d2t_path: str, t2d_path: str):
+        token_freq_dict = torch.load(
+            Path(token_freq) / "token_freq.pt", weights_only=True
+        )
+        d2t, t2d = build_vocab_mappings_from_distribution(
+            token_freq_dict=token_freq_dict,
+            draft_vocab_size=8192,
+            target_vocab_size=128256,
+        )
+
+        np.save(d2t_path, d2t.cpu().numpy())
+        np.save(t2d_path, t2d.cpu().numpy())
+
+    def _run_training(self, script_path: str, args_dict: dict):
+        cmd = [
+            "python",
```

Comment:
Maybe use `sys.executable` here (gives the absolute path to currently running python interpreter)?